jina/drivers/convert.py

```diff
@@ -136,8 +138,27 @@ def png_pack(png_tag, data):
 
         return 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
 
-    def convert(self, arr):
-        arr.uri = self.png_convertor(arr)
+    @staticmethod
+    def image_to_byte_array(image:Image, format:str):
+        import io
+        img_byte_arr = io.BytesIO()
+        image.save(img_byte_arr, format=format)
+        img_byte_arr = img_byte_arr.getvalue()
+        return img_byte_arr
+
+    def png_convertor_nd(self, arr: np.array):
```

Comment:
It is weird that we are saying that it is nd but we are converting and interpreting as an RGB image

Reply:
Better to say 3d and that is it

Reply:
I think it handles greyscale 2d but need to confirm it. Will change it to 3d for now.

Reply:
well then at least add exception if the shape is not properly handled and tested